clients/gitlabrepo/commits.go

```diff
@@ -27,31 +27,52 @@ import (
 )
 
 type commitsHandler struct {
-	glClient   *gitlab.Client
-	once       *sync.Once
-	errSetup   error
-	repourl    *repoURL
-	commitsRaw []*gitlab.Commit
+	glClient    *gitlab.Client
+	once        *sync.Once
+	errSetup    error
+	repourl     *repoURL
+	commitsRaw  []*gitlab.Commit
+	commitDepth int
 }
 
-func (handler *commitsHandler) init(repourl *repoURL) {
+func (handler *commitsHandler) init(repourl *repoURL, commitDepth int) {
 	handler.repourl = repourl
 	handler.errSetup = nil
 	handler.once = new(sync.Once)
+	handler.commitDepth = commitDepth
 }
 
 func (handler *commitsHandler) setup() error {
 	handler.once.Do(func() {
-		commits, _, err := handler.glClient.Commits.ListCommits(
-			handler.repourl.projectID,
-			&gitlab.ListCommitsOptions{
-				RefName: &handler.repourl.commitSHA,
```

Comment:
Would removing this `RefName` list option break the `--commit` option?

Reply:
Most likely, this was inadvertent. Will get it back in with the other changes